src/shared/install.c

```diff
@@ -2868,15 +2868,70 @@ static int read_presets(UnitFileScope scope, const char *root_dir, Presets *pres
         return 0;
 }
 
-static int query_presets(const char *name, const Presets presets) {
+static int pattern_match_multiple_instances(
+                        const char *pattern,
+                        const char *unit_name,
+                        char ***preset_array) {
+
+        const char *parameter;
+        char **iter;
+        _cleanup_free_ char *templated_name = NULL, *prefix = NULL, *instance_name = NULL;
+        int ret;
+
+        if (!preset_array || !unit_name_is_valid(unit_name, UNIT_NAME_TEMPLATE))
+                return -1;
```

Comment:
The return convention should be: 1 – matched, 0 – not matched, < 0 – an errno-style error. Don't return -1, because that means "permission denied".

But since this is a static function, an `assert` is more appropriate then returning an error.